tests/src/nonLinearElasticityTest.hh

```diff
@@ -148,3 +149,115 @@ auto NonLinearElasticityLoadControlNRandTR(const Material& mat) {
   t.check(gradTol >= nonLinOp.derivative().norm(), "Gradient Tolerance should be larger than actual tolerance");
   return t;
 }
+
+template <int gridDim, typename Material>
+auto GreenLagrangeStrainTest(const Material& mat) {
+  static_assert(gridDim == 2 or gridDim == 3, "Only supported for the 2D and 3D case");
+  TestSuite t("GreenLagrangeStrainTest with gridDim = " + std::to_string(gridDim));
+
+  /// For zero displacements, the stiffness matrix from geometrically linear and non-linear case should be the same,
+  /// there by ensuring correctness of the Green-Lagrange strain
+
+  const double tol = 1e-8;
+  auto grid        = createUGGridFromCorners<gridDim>(true);
+  auto gridView    = grid->leafGridView();
+
+  using namespace Dune::Functions::BasisFactory;
+  auto basis = Ikarus::makeBasis(gridView, power<gridDim>(lagrange<1>()));
+
+  std::vector<Ikarus::NonLinearElasticityFE<decltype(basis), decltype(mat)>> fes;
+  std::vector<Ikarus::LinearElastic<decltype(basis)>> fesLE;
+  for (auto& element : elements(gridView)) {
+    fes.emplace_back(basis, element, mat, nullptr, nullptr, nullptr);
+    fesLE.emplace_back(basis, element, 1000.0, 0.0, nullptr, nullptr, nullptr);
+  }
+
+  auto basisP = std::make_shared<const decltype(basis)>(basis);
+  Ikarus::DirichletValues dirichletValues(basisP->flat());
```

Comment:
since this test works with a single element you do not need the assemblers. just create the signle elements and call `KNonLin = calculateMatrix(...)`.